orchagent/neighorch.h

```diff
@@ -7,42 +7,68 @@
 #include "intfsorch.h"
 
 #include "ipaddress.h"
+#include "tokenize.h"
 
 #define NHFLAGS_IFDOWN                  0x1 // nexthop's outbound i/f is down
 
-struct NeighborEntry
+struct NextHopKey
```

Comment:
To me it looks better to be named as `NeighborKey `than `NextHopKey`.

Reply:
Now NeighborEntry is same as NextHopKey, I think NeighborEntry can be named as NeighborKey, but I haven't changed it. It is better to distinguish nexthop and neighbor, in some cases just need nexthop although now they are created together.

Reply:
My thought is, it is Neighorch and this is the key structure. So with NeighborKey, it aligns with the neighbor orch functionality. Nexthop is created _from_ the Neighbor. Just a suggestion to think, not a blocker

Reply:
I agree with Prince to replace the `NextHopKey` with `NeighborKey` and also remove the `NeighborEntry`. Otherwise it is confusing to have various class names.

Reply:
Also, since the two classes `NextHopKey` and `NextHopGroupKey` are very large. Please move them outside the header and create a separate header file and the implementation file.

Reply:
I think  `NexthopKey` semantic is more than `NeighborKey`.  A Route has  nexthop but not neigh. Neigh can reuse `NexthopKey` as `NeighborKey`. But route can't use `NeighborKey` as nexthop.
Spliting `Nexthopkey` and `NextHopGroupKey`  is good idea. We will fix it soon.